homeassistant/components/compit/climate.py

```diff
@@ -0,0 +1,331 @@
+"""Module contains the CompitClimate class for controlling climate entities."""
+
+from typing import Any
+
+from compit_inext_api import Device, Parameter
+
+from homeassistant.components.climate import (
+    ClimateEntity,
+    ClimateEntityFeature,
+    HVACMode,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import UnitOfTemperature
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, MANURFACER_NAME
+from .coordinator import CompitDataUpdateCoordinator
+
+type CompitConfigEntry = ConfigEntry[CompitDataUpdateCoordinator]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: CompitConfigEntry,
+    async_add_devices: AddEntitiesCallback,
+) -> None:
+    """Set up the CompitClimate platform from a config entry."""
+
+    coordinator: CompitDataUpdateCoordinator = entry.runtime_data
+
+    async_add_devices(
+        [
+            CompitClimate(
+                coordinator,
+                device,
+                device_definition.parameters,
+                device_definition.name,
+            )
+            for gate in coordinator.gates
```

Comment:
can we use `coordinator.devices` directly instead?

Reply:
No these list are different. One of the contains definitions but second contains devices returned from the api.

Reply:
in that case we should rename `coordinator.devices` to `coordinator.device_instances` or something, because otherwise it looks like it contains the actual devices